core/primitives/src/utils.rs

```diff
@@ -85,6 +85,15 @@ pub fn is_valid_top_level_account_id(account_id: &AccountId) -> bool {
         && VALID_TOP_LEVEL_ACCOUNT_ID.is_match(account_id)
 }
 
+/// Returns true if the account ID length is 64 characters and it's a hex representation.
+pub fn is_account_id_64_len_hex(account_id: &AccountId) -> bool {
+    account_id.len() == 64
+        && account_id.as_bytes().iter().all(|&b| match b {
```

Comment:
why do you need to convert to bytes first?

Reply:
Otherwise we'd iterate over `Char`s